### PR TITLE
#fixes user hierarchy-changes, activity-sensor spec fix in listing

### PR DESCRIPTION
**Pull request by Linoy339**
- user hierarchy changes in tag and credential
-  fixes in activity and sensor spec listing

### CODE REVIEW

File: src/repository/couch/CredentialRepository.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export class CredentialRepository implements CredentialInterface {
     }))
   }
   public async _insert(type_id: string | null, credential: any): Promise<{}> {
-    if (credential.origin === "me") {
+    if (credential.origin === "me" || credential.origin === "null") {
```
Comment (avaidyam) [inline src/repository/couch/CredentialRepository.ts:30]: **Please remove this.** `null !== "null"`, and we do not support a string containing the value `"null"` for this. Furthermore, both `type_id` and `origin` should be null here.

**Comment by avaidyam** [comment]
@Linoy339 Please remove every `"null"` and replace it with `null`. Null is a value, not a string.

**Comment by Linoy339** [comment]
@avaidyam .Understood. Origin will be sent as null and not "null". I will replace

Do the below also need to be changed to null ? From TypeService, the param type_id is not parsing  as null. 

https://github.com/BIDMCDigitalPsychiatry/LAMP-server/blob/user-hierarchy-changes-tag/src/repository/mongo/TypeRepository.ts#L115

**Comment by avaidyam** [comment]
Yes, every instance of string null needs to be replaced with the actual null in the Repository layer. The Service layer code needs to check if the value is string null and replace it with object null. (This is because a URL can only be a string so the service layer needs to sanitize the parameters and convert/replace them.)

**Comment by Linoy339** [comment]
Okay. Understood

**Comment by Linoy339** [comment]
@avaidyam . Please find the changes

**Comment by avaidyam** [comment]
It seems okay now. Please merge and test it in staging.

**Comment by Linoy339** [comment]
Sure @avaidyam . Will do

**Review by ZCOEngineer — APPROVED** [review]
Fixes user hierarchy-changes, activity-sensor spec fix in listing

**Pull request merged**